dtfs-central-api/src/services/changeStream/setupChangeStream.ts

```diff
@@ -27,10 +27,15 @@ const setupChangeStreamForDeletionCollection = async () => {
   const databaseConnection = await mongoDbClient.getConnection();
   const changeStream = databaseConnection
     .collection(MONGO_DB_COLLECTIONS.DELETION_AUDIT_LOGS)
-    .watch([{ $match: { operationType: 'insert' } }, { $project: { _id: 1, fullDocument: 1, ns: 1, documentKey: 1 } }], {
+    .watch([{ $match: { operationType: { $in: ['insert', 'update', 'replace'] } } }, { $project: { _id: 1, fullDocument: 1, ns: 1, documentKey: 1 } }], {
       fullDocument: 'updateLookup',
     });
   changeStream.on('change', (changeStreamDocument: ChangeStreamInsertDocument<DeletionAuditLog>) => {
+    // Deletion audit logs should never be updated or replaced. Cosmos DB does not currently support filtering at the `watch` step
+    // (See here https://learn.microsoft.com/en-us/answers/questions/356668/how-to-get-inserted-change-stream-data-use-cosmosd)
+    if (changeStreamDocument.operationType !== 'insert') {
+      throw new Error('A document in deletion-audit-logs has been %sd', changeStreamDocument.operationType);
```

Comment:
I don't think this works the way you want (because it's not a `console.error`)

Reply:
oops